File: frequentWords.py
from collections import defaultdict

#returns the first k most frequent words in a string(same frequency ordered alphabetically)
def frequentWords(words, k):

	d = defaultdict(int)

	for word in words.split(" "):
		word = word.lower().strip(",!_+=-@")

		d[word] += 1

	for v in sorted(d.items(), key = lambda kv: (-kv[1], kv[0])):
		print(v)
	
	output = [v[0] for v in sorted(d.items(), key = lambda kv: (-kv[1], kv[0]), reverse=False)]

	print(output)
	print(output[:k])


a = "apple apple apple azdy azdy azdy Gabba gabba hey, gabba gabba hey!"

frequentWords(a,2)

File: frequentWords.cpp
#include <iostream>
#include <cstring>
#include <sstream>
#include <algorithm>
#include <map>

using namespace std;

//My compiler is old, this builds on online CPP compiler

bool cmp(pair<string, int>& a, pair<string, int>& b)
{
	if (a.second != b.second) 
	{ 
		return a.second > b.second; 
	} 

	return a.first < b.first; 
}

string stripSymbols(string word)
{
	char chars[] = ",!_+=-@";

	for(unsigned int i = 0; i < strlen(chars); i++)
	{
		word.erase(remove(word.begin(), word.end(), chars[i]), word.end());
	}

	return word;
}

void frequentWords(string input, int numWords, string output[])
{
	stringstream ss(input);
	string buf;
	map<string, int> map;
	vector<pair<string, int>>v;
	

	while(ss >> buf)
	{
		transform(buf.begin(), buf.end(), buf.begin(), ::tolower);
		buf = stripSymbols(buf);
		map[buf]++;
	}



	for (auto itr = map.begin(); itr != map.end(); itr++)
	{
		cout << itr->first << ": " << itr->second << endl;
		v.push_back(*itr);
	}

	sort(v.begin(), v.end(), cmp);

	int i = -1;

	for(auto itr = v.begin(); itr != v.end(); itr++)
	{	
	    i++;
	    output[i] = itr->first;
	    
	    if(i >= numWords-1)
	        break;
	}
	
	for(i = 0; i < numWords; i++)
	cout << output[i] << endl;
} 


int main()
{
	string a = "apple apple apple azdy azdy azdy Gabba gabba hey, gabba gabba hey!";
	int freqWords = 2;
	string output[freqWords];

	frequentWords(a,freqWords, output);

	return 0;
}
